src/helics/application_api/Federate.cpp

```diff
@@ -1149,24 +1156,38 @@ CloningFilter &Federate::registerGlobalCloningFilter (const std::string &filterN
 
 void Federate::addSourceTarget (const Filter &filt, const std::string &targetEndpoint)
 {
-    coreObject->addSourceTarget (filt.getHandle (), targetEndpoint);
+    if (coreObject)
+    {
```

Comment:
I think having an extra check here is good but should we not add an else statement with a warning or something if this is called on a non coreObject?  (This is present in more places)

Reply:
other than throwing an error I am not sure what else to do.  The cases where that might be true are when an empty Federate somehow got created and used or after a call to finalize.  I could throw the invalid function call error?

Reply:
I think that is a good idea. It is definitely an edge case but I think an error trow is appropriate.

Reply:
there are probably a few more cases where I need to check it but I will fix them as I see them.